sealedenumprocessor/src/test/kotlin/com/livefront/sealedenum/SealedEnumWithAbstractBaseClassesTests.kt

```diff
@@ -0,0 +1,92 @@
+package com.livefront.sealedenum
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import kotlin.reflect.KTypeProjection
+import kotlin.reflect.full.createType
+import kotlin.reflect.full.isSubtypeOf
+
+interface BaseClassInterface1<T>
+
+interface BaseClassInterface2<T>
+
+interface BaseClassInterface3<out T>
+
+abstract class AlphaBase<A> : BaseClassInterface1<A>
+
+abstract class BetaBase<A, B : Any> : AlphaBase<A>(), BaseClassInterface2<B>
+
+@GenSealedEnum(generateEnum = true)
+sealed class SealedEnumWithAbstractBaseClasses : BetaBase<Any?, String>()
+
+abstract class GammaBase<out T : Any> : BaseClassInterface3<BaseClassInterface3<T>>
+
+@GenSealedEnum(generateEnum = true)
+sealed class SealedEnumWithAbstractBaseClassesCovariantType<out T : Any> : GammaBase<T>()
+
+class SealedEnumWithAbstractBaseClassesTests {
+    @Test
+    fun `enum implements correct interfaces with type arguments`() {
+        assertTrue(
+            SealedEnumWithAbstractBaseClassesEnum::class.createType().isSubtypeOf(
```

Comment:
`sealedenum` only depends on the Kotlin stdlib, which is the only consumer runtime.

`sealedenumprocessor` depends on `kotlin-reflect`, since `kotlinpoet` depends on it, but that is only included during annotation processing with `kapt`.